crates/oxc_linter/src/rules/unicorn/prefer_structured_clone.rs

```diff
@@ -0,0 +1,213 @@
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let config = value.get(0);
+
+        let allowed_functions = config
+            .and_then(|config| config.get("functions"))
+            .and_then(serde_json::Value::as_array)
+            .map(|v| {
+                v.iter().filter_map(serde_json::Value::as_str).map(ToString::to_string).collect()
+            })
+            .unwrap_or(vec![String::from("cloneDeep"), String::from("utils.clone")]);
+
+        Self(Box::new(PreferStructuredCloneConfig { allowed_functions }))
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr): oxc_ast::AstKind<'a> = node.kind() else {
+            return;
+        };
+
+        if call_expr.arguments.len() != 1 {
+            return;
+        }
+
+        if call_expr.optional {
+            return;
+        }
+
+        // `JSON.parse(JSON.stringify(…))
+        if is_method_call(call_expr, Some(&["JSON"]), Some(&["parse"]), Some(1), Some(1)) {
+            let Some(first_argument) = call_expr.arguments[0].as_expression() else {
+                return;
+            };
+
+            let Expression::CallExpression(inner_call_expr) =
+                first_argument.without_parenthesized()
+            else {
+                return;
+            };
+
+            if inner_call_expr.optional {
+                return;
+            }
+
+            if !is_method_call(
+                inner_call_expr,
+                Some(&["JSON"]),
+                Some(&["stringify"]),
+                Some(1),
+                Some(1),
+            ) {
+                return;
+            }
+
+            if inner_call_expr.arguments[0].is_spread() {
+                return;
+            }
+
+            let span = Span::new(call_expr.span.start, inner_call_expr.span.end);
+            ctx.diagnostic(prefer_structured_clone_diagnostic(span));
+        } else if !call_expr.arguments[0].is_spread() {
+            for function in &self.allowed_functions {
+                if let Some((object, method)) = function.split_once('.') {
+                    if is_method_call(call_expr, Some(&[object]), Some(&[method]), None, None) {
+                        ctx.diagnostic(prefer_structured_clone_diagnostic(call_expr.span));
+                    }
+                } else if is_method_call(call_expr, None, Some(&[function]), None, None)
+                    || is_method_call(call_expr, Some(&[function]), None, None, None)
+                    || is_specific_function_call(call_expr, function)
+                {
+                    ctx.diagnostic(prefer_structured_clone_diagnostic(call_expr.span));
+                }
+            }
+        }
+    }
+}
+
+fn is_specific_function_call(call_expr: &CallExpression, function_name: &String) -> bool {
+    if let Expression::Identifier(ident) = &call_expr.callee {
+        return ident.name == function_name;
+    }
```

Comment:
you can use the `is_specific_id` functin on callee? it handles a few more cases

https://github.com/oxc-project/oxc/blob/1365feb6b1409d5fd0de459180e0482cf52f4bb2/crates/oxc_ast/src/ast_impl/js.rs#L185-L190

Reply:
Done!